recipes/Tedlium2/Tokenizer/tedlium2_prepare.py

```diff
@@ -0,0 +1,184 @@
+"""
+Download link: https://lium.univ-lemans.fr/ted-lium2/
+Shucong Zhang 2023
+"""
+
+import os
+import csv
+import logging
+import torch
+import torchaudio
+import multiprocessing
+from speechbrain.dataio.dataio import merge_csvs
+
+logger = logging.getLogger(__name__)
+
+def clip_and_make_csv(sph_file, stm_file, utt_save_folder, csv_save_folder):
+    """
+    This function splits the .sph Ted-talk recording into utterences based on the .stm annotation
+
+    Arguments
+    ---------
+    sph_file : str
+        Path to the sph file containing Ted-talk recording.
+    stm_file : str
+        Path to the stm file containing Ted-talk annotation.
+    save_folder: str
+        The folder stores the clipped individual utterences.
+    enty: list
+        The list stores the csv entry
+    """
+    # the annotation for JillSobuleMANHATTANINJANUARY_2006.sph is not useful
+    if "JillSobuleMANHATTANINJANUARY_2006" in sph_file:
+        print("JillSobuleMANHATTANINJANUARY_2006.sph is skipped")
+        return
+
+    # load the annotation of the entire speech recording
+    annotation_file = open(stm_file, 'r')
+    annotations = annotation_file.readlines()
+
+    # load the original speech recording
+    original_speech, sample_rate = torchaudio.load(sph_file)
+
+    entry = []
+    # process the annotation utterence by utterence
+    for i, line in enumerate(annotations):
+        line = line.strip("\n")
+        line = line.split(" ")
+        # parse the annotation
+        talk_id = line[0]
+        spk_id = line[2]
+
+        # start and end point of the utterences in the recording
+        start = float(line[3])
+        end = float(line[4])
+        duration = - start + end
+        # we skip short utterences in case of CNN padding issues
+        if duration < 1.0:
```

Comment:
oof. I think it would be better to have it as an hyperparameter nop? It seems a bit too magical to me.